src/reducers/index.ts

```diff
@@ -1,17 +1,21 @@
-import ICounter from '../types';
+import { combineReducers } from 'redux';
+import { ICounter, LoadingStatus, IState } from '../types';
 import { ICounterAction, Actions } from '../actions';
 
-const initialState = [{ count: 0 }, { count: 5 }];
+const initialState: IState = {
```

Comment:
You should break out reducers by "feature", this helps keep them lightweight and easy to modify. So the API should be its own reducer because as it is right now separating out the counter logic and API handling logic is difficult.  (Multiple reducers can respond to the same action.)